File: grooves-core/src/main/java/com/github/rahulsom/grooves/EventType.java
package com.github.rahulsom.grooves;

/**
 * Represents the different types of events in the event sourcing system.
 */
public enum EventType {
    /** A standard event that applies changes to the aggregate. */
    Normal,
    /** An event that reverts a previous event. */
    Revert,
    /** An event that marks other events/aggregates as deprecated. */
    Deprecates,
    /** An event that indicates this event is deprecated by another. */
    DeprecatedBy
}

File: grooves-diagrams/src/main/kotlin/com/github/rahulsom/grooves/asciidoctor/EventType.kt
package com.github.rahulsom.grooves.asciidoctor

/**
 * Classifies types of events.
 *
 * @author Rahul Somasunderam
 */
enum class EventType {
    Normal,
    Revert,
    Deprecates,
    DeprecatedBy,
    Join,
    Disjoin,
}